src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php

```diff
@@ -0,0 +1,376 @@
+        // If this is a multi-site, include a host option so Lando imports to the correct database.
+        $hostOption = $siteName !== 'default' ? "--host=$siteName" : '';
+        $this->taskExec('lando')
+            ->arg('db-import')
+            ->arg($dbPath)
+            ->arg($hostOption)
+            ->run();
+
+        $this->say("Deleting $dbPath");
+        $this->taskExec('rm')->args($dbPath)->run();
+        return $this->drushDeployLando($siteName);
+    }
+
+    /**
+     * Refresh database on Tugboat.
+     *
+     * @return \Robo\Result
+     *   The result of the set of tasks.
+     */
+    public function databaseRefreshTugboat(): Result
+    {
+        $this->io()->title('refresh tugboat databases.');
+        $result = null;
+        foreach ($this->getAllSitesConfig() as $siteName => $siteInfo) {
+            try {
+                $dbPath = $this->databaseDownload($siteName);
+            } catch (TaskException $e) {
+                $this->yell("$siteName: No database configured. Download/import skipped.");
+                // @todo: Should we run a site-install by default?
+                continue;
+            }
+            if (empty($dbPath)) {
+                $this->yell("'$siteName' database path not found.");
+                continue;
+            }
+            $dbName = $siteName === 'default' ? 'tugboat' : $siteName;
+            $result = $this->taskExec('mysql')
+                ->option('-h', 'mariadb')
+                ->option('-u', 'tugboat')
+                ->option('-ptugboat')
+                ->option('-e', "drop database if exists $dbName; create database $dbName;")
+                ->run();
+
+            $this->io()->section("import $siteName database.");
+            $result = $this->taskExec("zcat $dbPath | mysql -h mariadb -u tugboat -ptugboat $dbName")
+                ->run();
+            $result = $this->taskExec('rm')->args($dbPath)->run();
+        }
+        return $result;
+    }
+
+    /**
+     * Generate Drupal login link.
+     *
+     * @param string $siteDir
+     *   The Drupal site directory name.
+     * @param bool $lando
+     *   Use lando to call drush, else call drush directly.
+     *
+     * @aliases uli
+     *
+     * @return \Robo\Result
+     *   The result of the set of tasks.
+     */
+    public function drupalLoginLink($siteDir = 'default', $lando = true): Result
+    {
+        $this->io()->section("create login link.");
+        if ($lando) {
+            $uri = $this->landoUri($siteDir);
+            $this->say("Lando URI detected: $uri");
+            return $this->taskExec('lando')
+                ->arg('drush')
+                ->arg('user:login')
+                ->option('--uri', $uri)
+                ->dir("$this->drupalRoot/sites/$siteDir")
+                ->run();
+        }
+        return $this->taskExec('../../../drush')
+            ->arg('user:login')
+            ->dir("$this->drupalRoot/sites/$siteDir")
+            ->run();
+    }
```

Comment:
Honest question: Given our lando and robo usage, is it reasonable to assume drush will always be at this hardcoded path?

Reply:
Great catch. This feels weird. Why is it not calling `drush` in `vendor/bin`?

Reply:
I learned something new with the solution. That `DrupalFinder` class is magic. 🔮 

@markdorison are you still setup to test this easily, or should I?

Reply:
@adamzimmermann I am booked up at the moment. If you can, fire away.

Reply:
A couple thoughts, though they may not be relevant.

1. When I look at the database refresh pattern there is one function for using lando, and one for using tugboat. Would it be better to follow that pattern and write two functions here?

2. Or, as long as we are willing to default to using lando, you could either:
  a. check if lando exists and use lando if it does 
  b. `try{ /lando implementation/} catch { try {/non-lando implementation}} finally{/uh oh/}`

Reply:
Did you mean to leave this comment elsewhere? I don't think it is related to this thread.

1. Yes, but I removed that line.
2. I'm not sure what you are getting at here, but I think the answer is no.
3. That's a good question. I would defer to @markdorison since he authored most of this.
4. The thing is Lando might exist and work, but that doesn't mean we want to use it, but perhaps we should just make that assumption and if someone wants a login link they can do it the old school `drush uli` route and then we can just get rid of the non-Lando path in the code.
5. Good question. I'm not sure.

I think answering that helped! Now we just need to see if @markdorison is on board now with ditching Lando support in the login robo command.

Reply:
@adamzimmermann I don't know how my comment ended up here. I was trying to respond to a recent commit/question you posted to us regarding how to get the login working in both lando and without lando. I will move my comment down to where it's supposed to be. sorry.